File: app/src/main/java/com/belous/v/clrc/view/dialog/ContextDialog.java
package com.belous.v.clrc.view.dialog;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;

import com.belous.v.clrc.R;
import com.belous.v.clrc.model.Item;

public class ContextDialog extends BaseDialog {

    public ContextDialog() {
    }

    public ContextDialog(Item item) {
        super(item);
    }

    @SuppressWarnings("ConstantConditions")
    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(context)
                .setTitle(R.string.select)
                .setItems(new String[]{getString(R.string.rename), getString(R.string.delete)}, (dialog, which) -> {
                    if (which == 0) {
                        new RenameDialog(item).show(getFragmentManager(), null);
                    } else {
                        new DeleteDialog(item).show(getFragmentManager(), null);

                    }
                })
                .create();
    }
}

File: app/src/main/java/com/belous/v/clrc/ui/component/ContextDialog.kt
package com.belous.v.clrc.ui.component

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.belous.v.clrc.R

class ContextDialog : DialogFragment() {

    companion object {
        const val YEELIGHT_ID = "yeelight_id"
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(requireContext())
            .setTitle(R.string.select)
            .setItems(
                arrayOf(getString(R.string.rename), getString(R.string.delete))
            ) { _: DialogInterface?, which: Int ->
                if (which == 0) {
                    RenameDialog().apply {
                        arguments = this@ContextDialog.arguments
                    }.show(parentFragmentManager, null)
                } else {
                    DeleteDialog().apply {
                        arguments = this@ContextDialog.arguments
                    }.show(parentFragmentManager, null)
                }
            }
            .create()
    }
}